app/models/miq_search.rb

```diff
@@ -41,6 +41,21 @@ def self.visible_to_all
     where("search_type=? or (search_type=? and (search_key is null or search_key<>?))", "global", "default", "_hidden_")
   end
 
+  def self.visible_to_current_user
+    where(:search_type => 'user', :search_key => User.current_user.userid)
+  end
+
+  def self.filters_by_type(type)
+    case type
+    when "global" # Global filters
+      visible_to_all
+    when "my"     # My filters
+      visible_to_current_user
+    else
+      raise "Error: #{type} is not a proper type!"
```

Comment:
I am not sure if we need an error message there, I considered it to be a good idea.. to think about all the cases that could occur somehow in the future..
I think, one shouldn't rely on the fact that maybe we don't need it. We should be sure if we need it or not. Maybe you or anybody else knows for sure.. 
If we need it, I can improve the error message.